### ISSUE TITLE
Money.to_integer/1

### ISSUE THREAD

**Issue by Schultzer** [kipcole9/money#42]
I propose a `to_integer/1` function which would take a `Money.t` and return the subunit in integers
this function would make it easier to use with services like stripe where its expected that the amount is represented in the subunit.

```elixir
iex> Money.new(:USD, "20.00") |> Money.to_integer()
2000
```

As far as I understand it we only need to take the coefficient of the `Decimal.t` so a function would maybe look like this

```elixir
def to_integer(%Money{amount: %Decimal{coef: subunit}}) do: subunit
```

what are your thoughts on this.?
let me know if you want me to create a pr.

**Comment by kipcole9** [comment]
Makes sense.  Would you consider a slight variation though?
```
{currency, integer, exp} = Money.to_integer_exp(money)
```
That way the explicitness of that fact that not all currencies have two digits is preserved and the user has to explicitly then discard it, as with the currency.  Lastly, and maybe its my paranoia, but theres a risk that any remainder after rounding is hidden or lost so I wonder if that should be returned too.

The implementation would need to also consider rounding and "reducing" (in Decimal terms) so something like:

```
  def to_integer_exp(%Money{} = money) do
    new_money = 
      money
      |> Money.round
      |> Money.reduce

   {:ok,  remainder} = Money.sub(money, new_money)
    {new_money.currency, new_money.amount.coef, new_money.amount.exp, remainder}
  end
```
This should guarantee the right rounding to the right number of decimal digits, conform the coef and exp and then return the components.

## Example
```
iex> m = Money.new(:USD, 200.012356)
#Money<:USD, 200.012356>
iex> Money.to_integer_exp(m)        
{:USD, 20001, -2, #Money<:USD, 0.002356>}
```
I've put an initial implemtation in master since its a pretty simple thing and I can adjust the code if you have a different view.

**Comment by Schultzer** [comment]
This is perfect just the stuff I was after 😄 ❤️ 
and way more clever than my solutions.

**Comment by kipcole9** [comment]
Published on [hex as version 1.1.1](hex.pm/packages/ex_money).
